app/redux/ducks/interventions.js

```diff
@@ -46,43 +50,81 @@ export function unsubscribe(channel) {
   return { type: UNSUBSCRIBE, payload: channel };
 }
 
-export function notify(notification) {
-  const { message } = notification.data;
-  const { type } = message;
-  if (type === 'subject_queue') {
-    const subjectIDs = message.subject_ids;
-    const workflowID = message.workflow_id;
-    return (dispatch) => {
+
+function prependSubjectQueue(data) {
+  const subjectIDs = data.subject_ids;
+  const workflowID = data.workflow_id;
+  return (dispatch) => {
+    dispatch({
+      type: FETCH_SUBJECTS,
+      payload: subjectIDs
+    });
+    apiClient.type('subjects').get(subjectIDs)
+    .catch((error) => {
       dispatch({
-        type: FETCH_SUBJECTS,
-        payload: subjectIDs
-      });
-      apiClient.type('subjects').get(subjectIDs)
-      .catch((error) => {
-        dispatch({
-          type: ERROR,
-          payload: error
-        });
-        return [];
-      })
-      .then(subjects => subjects.map((subject) => {
-        subject.update({ 'metadata.intervention': true });
-        return subject;
-      }))
-      .then((subjects) => {
-        dispatch(prependSubjects(subjects, workflowID));
+        type: ERROR,
+        payload: error
       });
-    };
-  } else {
-    return { type: ADD_NOTIFICATION, payload: notification };
+      return [];
+    })
+    .then(subjects => subjects.map((subject) => {
+      subject.update({ 'metadata.intervention': true });
+      return subject;
+    }))
+    .then((subjects) => {
+      dispatch(prependSubjects(subjects, workflowID));
+    });
+  };
+};
+
+export function intervention(message) {
```

Comment:
I prefer function names to be verbs rather than nouns. Would `dispatchIntervention` be a better name for this action creator? `handleMessage` might work too, but is kind of generic.

Reply:
changed to `processIntervention(message)` to avoid double dispatch wording in `main.cjsx`